anaDACScan.py

```diff
@@ -0,0 +1,345 @@
+        print("Error: unexpected value of nameY: '%s'"%nameY)
+        exit(1)
+
+    from utils.anaInfo import nominalDacValues
+    from utils.anaInfo import nominalDacScalingFactors
+        
+    if nameX not in nominalDacValues.keys():
+        print("Error: unexpected value of nameX: '%s'"%nameX)
+        exit(1)
+
+    #the nominal reference current is 10 uA and it has a scaling factor of 0.5   
+    nominal_iref = 10*0.5
+
+    nominal = nominalDacValues[nameX][0]
+
+    #convert all voltages to mV and currents to uA
+    if nominalDacValues[nameX][1] == "V":
+        nominal *= pow(10.0,3)
+    elif nominalDacValues[nameX][1] == 'mV':
+        pass
+    elif nominalDacValues[nameX][1] == 'uV':
+        nominal *= pow(10.0,-3)
+    elif nominalDacValues[nameX][1] == 'nV':
+        nominal *= pow(10.0,-6)
+    elif nominalDacValues[nameX][1] == "A":
+        nominal *= pow(10.0,6)
+    elif nominalDacValues[nameX][1] == 'mA':
+        nominal *= pow(10.0,3)
+    elif nominalDacValues[nameX][1] == 'uA':
+        pass
+    elif nominalDacValues[nameX][1] == 'nA':
+        nominal *= pow(10.0,-3)
+    else:
+        print("Error: unexpected units: '%s'"%nominalDacValues[nameX][1])
+        exit(1)
+
+    calInfo = {}
+
+    if args.calFileList != None:
+        for line in open(args.calFileList):
+            if line[0] == "#":
+                continue
+            line = line.strip(' ').strip('\n')
+            first = line.split(' ')[0].strip(' ')
+            second = line.split(' ')[1].strip(' ')
+            tuple_calInfo = parseCalFile(second)
+            calInfo[int(first)] = {'slope' : tuple_calInfo[0], 'intercept' : tuple_calInfo[1]}
+
+    #for each OH, check if calibration files were provided, if not search for the calFile in the $DATA_PATH, if it is not there, then skip that OH for the rest of the script
+    for oh in ohArray:
+        if oh not in calInfo.keys():
+            calAdcCalFile = "{0}/{1}/calFile_{2}_{1}.txt".format(dataPath,chamber_config[oh],nameY)
+            calAdcCalFileExists = os.path.isfile(calAdcCalFile)
+            if not calAdcCalFileExists:
+                print("Skipping OH{0}, detector {1}, missing CFG_CAL_DAC Calibration file:\n\t{2}".format(
+                    oh,
+                    chamber_config[oh],
+                    calAdcCalFile))
+                ohArray = np.delete(ohArray,(oh))
+
+    if len(ohArray) == 0:
+        print('No OHs with a calFile, exiting.')
+        exit(1)
+           
+    # Initialize nested dictionaries
+    dict_DACvsADC_Graphs = nesteddict()
+    dict_RawADCvsDAC_Graphs = nesteddict()
+    dict_DACvsADC_Funcs = nesteddict()
+
+    # Initialize a TGraphErrors and a TF1 for each vfat
+    for oh in ohArray:
+        for vfat in range(0,24):
+            dict_RawADCvsDAC_Graphs[oh][vfat] = r.TGraphErrors()
+            dict_RawADCvsDAC_Graphs[oh][vfat].GetXaxis().SetTitle(nameX)
+            dict_RawADCvsDAC_Graphs[oh][vfat].GetYaxis().SetTitle(nameY)
```

Comment:
to assist with reading the 3x8 grid plots (especially for people who are not familiar with the chip layout), please add:

```python
dict_RawADCvsDAC_Graphs[oh][vfat].SetTitle("VFAT{}".format(vfat))
```

For all plots.

Reply:
done

Reply:
doesn't seem like the change is reflected here.

Reply:
the 3x8 grid is for the calibrated, not raw, plots

see this commit: https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/158/commits/8974a78b647e25177e7e5461beead27bfbcca53e

Reply:
yes I see, I thought a 3x8 plot was made for each distribution